src/services/sync-component-with-file/find-config-file-changes.ts

```diff
@@ -19,62 +20,94 @@ const getRemovedFiles = async (
   compassYmlFilesDiffs: CommitFileDiff[],
   event: PushEvent,
 ): Promise<ComponentUnlinkPayload[]> => {
-  return Promise.all(
+  const settledPromises = await Promise.allSettled(
     compassYmlFilesDiffs.map((diff: CommitFileDiff) => {
-      return getFileContent(token, event.project.id, diff.old_path, event.before).then((componentYaml) => ({
-        componentYaml,
-        filePath: `/${diff.new_path}`,
-        immutableLocalKeyPrefix: event.project.id.toString(),
-      }));
+      return getFileContent(token, event.project.id, diff.old_path, event.before)
+        .then((componentYaml) => ({
+          componentYaml,
+          filePath: `/${diff.new_path}`,
+          immutableLocalKeyPrefix: event.project.id.toString(),
+        }))
+        .catch((err) => {
+          console.error(`Unable to get removed file. Error: ${err}`);
+        });
     }),
   );
+  return settledPromises
+    .filter((result) => result.status === 'fulfilled')
+    .map((result) => (result as PromiseFulfilledResult<ComponentUnlinkPayload>).value);
 };
 
 const getAddedFiles = async (
   token: string,
   compassYmlFilesDiffs: CommitFileDiff[],
   event: PushEvent,
 ): Promise<ComponentSyncPayload[]> => {
-  return Promise.all(
+  const settledPromises = await Promise.allSettled(
     compassYmlFilesDiffs.map((diff: CommitFileDiff) =>
-      getFileContent(token, event.project.id, diff.new_path, event.after).then((componentYaml) => ({
-        componentYaml,
-        absoluteFilePath: diff.new_path,
-        filePath: `/${diff.new_path}`,
-      })),
+      getFileContent(token, event.project.id, diff.new_path, event.after)
+        .then((componentYaml) => ({
+          componentYaml,
+          absoluteFilePath: diff.new_path,
+          filePath: `/${diff.new_path}`,
+        }))
+        .catch((err) => {
+          console.error(`Unable to get added file. Error: ${err}`);
```

Comment:
@VladyslavLevinUA noted that this should error so that the settled Promises doesnt come back as empty. Ill fix this next week.

Reply:
now throwing